docs/book/src/reference/using_an_external_type.md

```diff
@@ -0,0 +1,155 @@
+
+Currently Kubebuilder handles the first two, CRDs and Core Resources, seamlessly.  External CRDs and CRs created via Aggregation must be scaffolded manually.
+
+In order to use a Kubernetes Custom Resource that has been defined in another project
+you will need to have several items of information.
+* The Domain of the CR
+* The Group under the Domain 
+* The Go import path of the CR Type definition.
+
+The Domain and Group variables have been discussed in other parts of the documentation.  The import path would be located in the project that installs the CR.
+
+This document uses `my` and `their` prefixes as a naming convention for repos, groups, and types to clearly distinguish between your own project and the external one you are referencing.
+
+Example external API Aggregation directory structure
+```
+github.com
+    ├── theiruser
+        ├── theirproject
+            ├── apis
+                ├── theirgroup
+                   ├── doc.go
+                   ├── install
+                   │   ├── install.go
+                   ├── v1alpha1
+                   │   ├── doc.go
+                   │   ├── register.go
+                   │   ├── types.go
+                   │   ├── zz_generated.deepcopy.go
+```
+
+In the case above the import path would be `github.com/theiruser/theirproject/apis/theirgroup/v1alpha1`
+
+### Create a project
+
+```shell
+kubebuilder init --domain $APIDOMAIN --owner "MyCompany"
+```
+
+### Add a controller
+
+be sure to answer no when it asks if you would like to create an api? [Y/n]
+```shell
+kubebuilder create api --group mygroup --version $APIVERSION --kind MyKind
+```
+
+## Edit the API files.
+
+### Register your Types
+
+Edit the following file to the pkg/apis directory to append their `AddToScheme` to your `AddToSchemes`:
+
+file: pkg/apis/mytype_addtoscheme.go
+```go
+package apis
+
+import (
+	mygroupv1alpha1 "github.com/myuser/myrepo/apis/mygroup/v1alpha1"
+	theirgroupv1alpha1 "github.com/theiruser/theirproject/apis/theirgroup/v1alpha1"
+)
+
+func init() {
+	// Register the types with the Scheme so the components can map objects 
+	// to GroupVersionKinds and back
+	AddToSchemes = append(
+	  AddToSchemes, 
+	  mygroupv1alpha1.SchemeBuilder.AddToScheme,
+	  theirgroupv1alpha1.SchemeBuilder.AddToScheme,
+	)
+}
+
+```
+
+## Edit the Controller files
+
+### Use the correct imports for your API
+
+file: pkg/controllers/mytype_controller.go
+```go
+package controllers
+
+import (
+	mygroupv1alpha1 "github.com/myuser/myrepo/apis/mygroup/v1alpha1"
+	theirgroupv1alpha1 "github.com/theiruser/theirproject/apis/theirgroup/v1alpha1"
+)
+```
+
+Note that core resources may simply be imported by depending on the API's from upstream Kubernetes:
+
+```go
+package controllers
+// contains core resources like Deployment
+import (
+   v1 "k8s.io/api/apps/v1"
+)
+```
+
+### Update dependencies
+
+```
+go mod tidy
+```
+
```

Comment:
I think we also need to recommend checking if the kind defined by the other project will be applied to the cluster.

```suggestion
### Use the Discovery API to see if the new API is present

The best approach at this point is to make sure that the resource is available in the cluster because we are using a third-party API. So that, when the manager initializes we should ensure that the external type required is applied to the cluster. 

#### Let's start by creating the discovery client:

```go
// Get a config to talk to the apiserver
    cfg, err := config.GetConfig()
    if err != nil {
        log.Error(err, "")
        os.Exit(1)
    }
    
    // Create the discoveryClient
    discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
    if err != nil {
        log.Error(err, "Unable to create discovery client")
        os.Exit(1)
    }
```

#### Now, we can search for all API resources as follows:

```go
    // Get a list of all API's on the cluster
    apiGroup, apiResourceList , err := discoveryClient.ServerGroupsAndResources()
    if err != nil {
        log.Error(err, "Unable to get Group and Resources")
        os.Exit(1)
    }
```

<aside class="note">
<h1>Checking the resources applied on the cluster with kubectl</h1>

Note that by using the kubectl api-resources command, it is possible to check the resources available in the cluster.

</aside>

#### Now, we can check and verify if the resource is or is not on the cluster:

```
    // Looking for the external Type
    for i := 0; i < len(apiResourceList); i++ {
        if  apiResourceList[i].Kind == kind {
            // found the Kind
        }
    }
```
Note: Check the [APIResourceList](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#APIResourceList) GoDoc to see other options. Also, see that you could either use [APIGroup](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#APIGroup)  where within is possible to use other attributes, such as the Group, Version, and Kind.

```

Reply:
IMO we should decide wether we want to recommend an in-code guide with interaction on startup (if the crd is there) or if we simply want to use kubectl here directly. After All we can just get the CRDs and verify their kind e.g. `kubectl api-resources --api-group=telemetry.istio.io` or `kubectl describe crds` instead of the incode guide. WDYT?

Reply:
Hi @jakobmoellersap,

Not sure if I properly follow up sorry. But I think in the suggestion we are using the discovery client and then checking if the Kind(CRD) is or not applied on the cluster. if not then, we return an error and kill the manager. Also, we are adding a note to let the users are able to check the docs and see all options.

However, please, feel free to shape it as you see fit. I just tried to provide an example of the suggestions and what I think it is missing.

Reply:
Maybe it makes sense to highlight that there are different ways to accomplish this rather than providing an explicit example and that it is up to the reader to implement what they feel is the best approach for their case?